nep-dapi.mediawiki

```diff
@@ -0,0 +1,1986 @@
+            "type": "Array",
+            "value": [
+              {
+                "type": "ByteArray",
+                "value": "7472616e73666572"
+              },
+              {
+                "type": "ByteArray",
+                "value": "296ac124021a71c449a9bad320c16429b08ad6ee"
+              },
+              {
+                "type": "ByteArray",
+                "value": "7869ef9732cdf6f6d54adaa5cae3b55a9396bceb"
+              },
+              {
+                "type": "ByteArray",
+                "value": "00e1f505"
+              }
+            ]
+          }
+        },
+        {
+          "contract": "0x849d095d07950b9e56d0c895ec48ec5100cfdff1",
+          "state": {
+            "type": "Array",
+            "value": [
+              {
+                "type": "ByteArray",
+                "value": "7472616e73666572"
+              },
+              {
+                "type": "ByteArray",
+                "value": "296ac124021a71c449a9bad320c16429b08ad6ee"
+              },
+              {
+                "type": "ByteArray",
+                "value": "b1fdddf658ce5ff9f83e66ede2f333ecfcc0463e"
+              },
+              {
+                "type": "ByteArray",
+                "value": "00e1f505"
+              }
+            ]
+          }
+        }
+      ]
+    }
+  ]
+}
+</pre>
+
+
+===Write Methods===
+====send====
+Invoke a transfer of a specified amount of a given asset from the connected account to another account.
+
+
+'''Method Interface'''
+
+<pre>
+function send({
+  fromAddress: string,
+  toAddress: string,
+  asset: string,
+  amount: string,
+  remark?: string,
+  fee?: string,
+  network?: string,
+  broadcastOverride?: boolean,
+}): Promise<SendOutput>
+</pre>
+
+
+'''Input arguments'''
+
+<pre>
+interface SendArgs {
+  fromAddress: string; // Address of the connected account to send the assets from
+  toAddress: string; // Address of the receiver of the assets to be sent
+  asset: string; // Asset script hash to be sent. Accepts asset symbol only for "MainNet"
+  amount: string; // The parsed amount of the asset to be sent
+  remark?: string; // (Optional) Description of the transaction to be made
+  fee?: string; // (Optional) The parsed amount of network fee (in GAS) to include with transaction
+  network?: string; //  Network to submit this request to. If omitted, will default to network the wallet is currently set to.
+  broadcastOverride?: boolean; // In the case that the dApp would like to be responsible for broadcasting the signed transaction rather than the wallet provider
+}
+</pre>
+
+
+'''Success return value'''
+
+<pre>
+interface SendOutput {
+  txid: string; // The transaction ID of the send invocation
+  nodeUrl?: string; // The node which the transaction was broadcast to. Returned if transaction is broadcast by wallet provider
+  signedTx?: string; // The serialized signed transaction. Only returned if the broadcastOverride input argument was set to True
```

Comment:
I don't really have any preference on this one, since a couple of wrapper methods could be created in the client side packages to leverage this single method.

If anyone else has a strong opinion on adding additional methods to this standard for send, invoke, and deploy. Where the main difference will be structure of the output based on if the method is broadcasting vs just signing. From a functionality standpoint they are both the same, and the user should already know based on their inputs what the output object structure should be. However I can understand some have a preference for very strict typing.

Reply:
For references, EOS standard lib uses the same invoke method (`transact`) for both broadcasting and not:
https://eosio.github.io/eosjs/reference/classes/api.api-1.html#transact

Both the serialized txn and signature is returned in both cases so maybe that is a possible interface?

I have no real preference regarding this as well though.